### PR TITLE
OSDOCS-4842: update MicroShift REST API documentation

### PR DESCRIPTION
**Pull request by dhellmann**
This PR imports newly generated REST API documentation files based on
the APIs present in MicroShift.

See https://github.com/openshift/microshift/pull/1320 for the tools and instructions used to produce this PR.

Version(s): 4.12, 4.13

Issue: [OSDOCS-4842](https://issues.redhat.com//browse/OSDOCS-4842)

Link to docs preview:

QE review:
  N/A, author is SME

### CODE REVIEW

**Comment by dhellmann** [comment]
The build is failing because some of the API type descriptions end with escaped double quotes, which the tool doesn't strip and asciidoctor doesn't re-escape, and which then cause the XML document created by asciidoctor to be mal-formed.

The openapi.json includes:

```
    "io.k8s.api.autoscaling.v1.CrossVersionObjectReference": {
      "description": "CrossVersionObjectReference contains enough information to let you identify the referred resource.",
      "type": "object",
      "required": [
        "kind",
        "name"
      ],
      "properties": {
        "apiVersion": {
          "description": "API version of the referent",
          "type": "string"
        },
        "kind": {
          "description": "Kind of the referent; More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds\"",
          "type": "string"
        },
        "name": {
          "description": "Name of the referent; More info: http://kubernetes.io/docs/user-guide/identifiers#names",
          "type": "string"
        }
      },
      "x-kubernetes-map-type": "atomic"
    },
```

which comes out in the asciidoc from openshift-apidoc-gen as

```
|===
=== .spec.metrics[].object.describedObject
Description::
+
--
CrossVersionObjectReference contains enough information to let you identify the referred resource.
--

Type::
  `object`

Required::
  - `kind`
  - `name`



[cols="1,1,1",options="header"]
|===
| Property | Type | Description

| `apiVersion`
| `string`
| API version of the referent

| `kind`
| `string`
| Kind of the referent; More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds"

| `name`
| `string`
| Name of the referent; More info: http://kubernetes.io/docs/user-guide/identifiers#names
```

which ends up in the DocBook XML in part as

```
<entry align="left" valign="top"><simpara>Kind of the referent; More info: <link xlink:href="https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds"">https://git.       k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds"</link></simpara></entry>
```

and that `""` at the end of the xlink:href attribute causes the parser error.

**Comment by dhellmann** [comment]
Adding a cleanup script to remove the trailing quotes from descriptions eliminated that error, but now I'm seeing lots of messages like:

```
>>> Working on microshift_rest_api book in microshift <<<
Transforming the AsciiDoc content to DocBook XML...
Unknown ID or title "io.k8s.apimachinery.pkg.apis.meta.v1.ListMeta", used as an internal cross reference
```

**Comment by dhellmann** [comment]
The missing references are for types that are defined in the openapi.json file but that do not make it into the microshift_rest_api/objects/index.adoc file.

```
$ grep '"io.k8s.apimachinery.pkg.apis.meta' _output/openshift-apidocs-gen/openapi.json _output/openshift-apidocs-gen/microshift_rest_api/objects/index.adoc
_output/openshift-apidocs-gen/openapi.json:    "io.k8s.apimachinery.pkg.apis.meta.v1.APIGroup": {
_output/openshift-apidocs-gen/openapi.json:    "io.k8s.apimachinery.pkg.apis.meta.v1.APIGroupList": {
_output/openshift-apidocs-gen/openapi.json:    "io.k8s.apimachinery.pkg.apis.meta.v1.APIResource": {
_output/openshift-apidocs-gen/openapi.json:    "io.k8s.apimachinery.pkg.apis.meta.v1.APIResourceList": {
_output/openshift-apidocs-gen/openapi.json:    "io.k8s.apimachinery.pkg.apis.meta.v1.APIVersions": {
_output/openshift-apidocs-gen/openapi.json:    "io.k8s.apimachinery.pkg.apis.meta.v1.Condition": {
_output/openshift-apidocs-gen/openapi.json:    "io.k8s.apimachinery.pkg.apis.meta.v1.DeleteOptions": {
_output/openshift-apidocs-gen/openapi.json:    "io.k8s.apimachinery.pkg.apis.meta.v1.FieldsV1": {
_output/openshift-apidocs-gen/openapi.json:    "io.k8s.apimachinery.pkg.apis.meta.v1.GroupVersionForDiscovery": {
_output/openshift-apidocs-gen/openapi.json:    "io.k8s.apimachinery.pkg.apis.meta.v1.LabelSelector": {
_output/openshift-apidocs-gen/openapi.json:    "io.k8s.apimachinery.pkg.apis.meta.v1.LabelSelectorRequirement": {
_output/openshift-apidocs-gen/openapi.json:    "io.k8s.apimachinery.pkg.apis.meta.v1.ListMeta": {
_output/openshift-apidocs-gen/openapi.json:    "io.k8s.apimachinery.pkg.apis.meta.v1.ManagedFieldsEntry": {
_output/openshift-apidocs-gen/openapi.json:    "io.k8s.apimachinery.pkg.apis.meta.v1.MicroTime": {
_output/openshift-apidocs-gen/openapi.json:    "io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta": {
_output/openshift-apidocs-gen/openapi.json:    "io.k8s.apimachinery.pkg.apis.meta.v1.OwnerReference": {
_output/openshift-apidocs-gen/openapi.json:    "io.k8s.apimachinery.pkg.apis.meta.v1.Patch": {
_output/openshift-apidocs-gen/openapi.json:    "io.k8s.apimachinery.pkg.apis.meta.v1.Preconditions": {
_output/openshift-apidocs-gen/openapi.json:    "io.k8s.apimachinery.pkg.apis.meta.v1.ServerAddressByClientCIDR": {
_output/openshift-apidocs-gen/openapi.json:    "io.k8s.apimachinery.pkg.apis.meta.v1.Status": {
_output/openshift-apidocs-gen/openapi.json:    "io.k8s.apimachinery.pkg.apis.meta.v1.StatusCause": {
_output/openshift-apidocs-gen/openapi.json:    "io.k8s.apimachinery.pkg.apis.meta.v1.StatusDetails": {
_output/openshift-apidocs-gen/openapi.json:    "io.k8s.apimachinery.pkg.apis.meta.v1.Time": {
_output/openshift-apidocs-gen/openapi.json:    "io.k8s.apimachinery.pkg.apis.meta.v1.WatchEvent": {
_output/openshift-apidocs-gen/microshift_rest_api/objects/index.adoc:[id="io.k8s.apimachinery.pkg.apis.meta.v1.Condition"]
_output/openshift-apidocs-gen/microshift_rest_api/objects/index.adoc:[id="io.k8s.apimachinery.pkg.apis.meta.v1.DeleteOptions"]
_output/openshift-apidocs-gen/microshift_rest_api/objects/index.adoc:[id="io.k8s.apimachinery.pkg.apis.meta.v1.LabelSelector"]
_output/openshift-apidocs-gen/microshift_rest_api/objects/index.adoc:[id="io.k8s.apimachinery.pkg.apis.meta.v1.MicroTime"]
_output/openshift-apidocs-gen/microshift_rest_api/objects/index.adoc:[id="io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta"]
_output/openshift-apidocs-gen/microshift_rest_api/objects/index.adoc:[id="io.k8s.apimachinery.pkg.apis.meta.v1.Patch"]
_output/openshift-apidocs-gen/microshift_rest_api/objects/index.adoc:[id="io.k8s.apimachinery.pkg.apis.meta.v1.Status"]
_output/openshift-apidocs-gen/microshift_rest_api/objects/index.adoc:[id="io.k8s.apimachinery.pkg.apis.meta.v1.Time"]
_output/openshift-apidocs-gen/microshift_rest_api/objects/index.adoc:[id="io.k8s.apimachinery.pkg.apis.meta.v1.WatchEvent"]
```

File: microshift_rest_api/index.adoc [hunk 1]
```diff
@@ -0,0 +1,119 @@
+// Automatically generated by 'openshift-apidocs-gen'. Do not edit.
+:_content-type: ASSEMBLY
+[id="api-index"]
+= API index
+
+[cols="1,1",options="header"]
+|===
+^| API ^| API group
+| xref:./extension_apis/apiservice-apiregistration-k8s-io-v1.adoc#apiservice-apiregistration-k8s-io-v1[APIService]
```
Comment (ShaunaDiaz) [inline microshift_rest_api/index.adoc:9]: All of these likely have to be in the form of link:https://; for example,
link:https://access.redhat.com/solutions/60959[How to embed a Kickstart file into an ISO image].

File: microshift_rest_api/metadata_apis/binding-v1.adoc [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+| `apiVersion`
+| `string`
+| APIVersion defines the versioned schema of this representation of an object. Servers should convert recognized schemas to the latest internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources
+
+| `kind`
+| `string`
+| Kind is a string value representing the REST resource this object represents. Servers may infer this from the endpoint the client submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+
+| `metadata`
+| xref:../objects/index.adoc#io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta[`ObjectMeta`]
```
Comment (ShaunaDiaz) [inline microshift_rest_api/metadata_apis/binding-v1.adoc:39]: Here, too.

**Comment by dhellmann** [comment]
I'll come back to this later.

**Pull request closed without merging**